lib/template-loader.php

```diff
@@ -195,6 +193,12 @@ function gutenberg_find_template( $template_file ) {
 	}
 
 	if ( $current_template_post ) {
+		if ( is_admin() ) {
+			foreach ( parse_blocks( $current_template_post->post_content ) as $block ) {
+				create_auto_draft_for_template_part_block( $block );
+			}
+		}
+		$_wp_current_template_id      = $current_template_post->ID;
```

Comment:
Why do we need to change the template resolution code in this PR?

Reply:
Because we need the current template ID saved, and to create template part auto drafts for custom saved templates as well as file ones.